databases/functions/topics/topic-get-topic-detail-by-id.js

```diff
@@ -0,0 +1,36 @@
+const {QueryTypes} = require('sequelize');
+
+const getTopicDetailById = async (sequelize, selfUserId, topicName) => {
+  const query = `SELECT 
+      topics.*, 
+      COUNT(user_topics.topic_id) as topic_follower_count, 
+      CASE COALESCE(user_following_topics.user_id, '') 
+        WHEN '' THEN false
+        ELSE true
+      END as is_following
+    FROM topics
+    LEFT JOIN user_topics
+      ON user_topics.topic_id = topics.topic_id
+    LEFT JOIN user_topics AS user_following_topics
+      ON user_following_topics.topic_id = topics.topic_id AND user_following_topics.user_id=:selfUserId
+    WHERE topics.name=:topicName
+    GROUP BY user_topics.topic_id, topics.topic_id, user_following_topics.user_id`;
+
+  try {
+    const response = await sequelize.query(query, {
+      type: QueryTypes.SELECT,
+      replacements: {
+        selfUserId,
+        topicName
+      }
+    });
+
+    return response?.[0];
+  } catch (error) {
+    console.error('Error on fetching topic detail');
+    console.error(error);
+    return null;
+  }
+};
+
+module.exports = getTopicDetailById;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `getTopicDetailById` function seems to be well implemented. It uses parameterized queries, which is a good practice to prevent SQL injection attacks. However, it's important to note that error handling could be improved. Currently, the function just logs the error and returns null. This might not be very helpful for the caller of this function as they won't know what went wrong. Instead, consider throwing the error or returning an error object so that the caller can handle it appropriately.

```diff
-    console.error('Error on fetching topic detail');
-    console.error(error);
-    return null;
+    throw error;
```
This way, the caller can decide how to handle the error (e.g., retry, log, show a message to the user, etc.).


<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
closed, since we just want to return null